dev/io.openliberty.microprofile.openapi.2.0.internal_fat/fat/src/io/openliberty/microprofile/openapi20/fat/deployments/StartupWarningMessagesTest.java

```diff
@@ -0,0 +1,135 @@
+
+        ServerConfiguration serverConfig = server.getServerConfiguration().clone();
+        serverConfig.addApplication("serverXMLNameIncluded", "testEarIncluded.ear", "ear");
+        serverConfig.addApplication("serverXMLNameExcluded", "testEarExcluded.ear", "ear");
+
+        MpOpenAPIElement.MpOpenAPIElementBuilder.cloneBuilderFromServerResetAppsAndModules(server)
+                                                //these match the application's deployment descriptor but not its name as set in server.xml.
+                                                //a warning should be emitted informing the user of their likely mistake and how o fix it.
+                                                .addIncludedApplicaiton("testEarIncluded")
+                                                .addExcludedApplicaiton("testEarExcluded")
+                                                .addIncludedModule("testEarIncluded/test1")
+                                                .addExcludedModule("testEarExcluded/test2")
+                                                .buildAndOverwrite(serverConfig.getMpOpenAPIElement());
+
+        server.updateServerConfiguration(serverConfig);
+
+        WebArchive war1 = ShrinkWrap.create(WebArchive.class, "test1.war")
+                                    .addClasses(DeploymentTestApp.class, DeploymentTestResource.class);
+
+        EnterpriseArchive ear = ShrinkWrap.create(EnterpriseArchive.class, "testEarIncluded.ear")
+                                          .addAsModules(war1);
+
+        WebArchive war2 = ShrinkWrap.create(WebArchive.class, "test2.war")
+                                    .addClasses(DeploymentTestApp.class, DeploymentTestResource.class);
+
+        EnterpriseArchive ear2 = ShrinkWrap.create(EnterpriseArchive.class, "testEarExcluded.ear")
+                                           .addAsModules(war2);
+
+        ShrinkHelper.exportAppToServer(server, ear, SERVER_ONLY, DISABLE_VALIDATION);//set up validation manually because the app name doesn't match the archive name
+        server.addInstalledAppForValidation("serverXMLNameIncluded");
+        ShrinkHelper.exportAppToServer(server, ear2, SERVER_ONLY, DISABLE_VALIDATION);
+        server.addInstalledAppForValidation("serverXMLNameExcluded");
+
+        server.startServer();
+
+        //Example message: CWWKO1680W: The testEar application name in the includeApplication or includeModule configuration element does not match the name of any deployed application but it does match the name from the deployment descriptor of the serverXMLName application. The application name used here must be the application name specified in server.xml, or the archive file name with the extension removed if no name is specified in server.xml.
+        List<String> messages = new ArrayList<>();
+        messages.add("CWWKO1680W: The testEarIncluded/test1 application name in the includeModule configuration.*serverXMLNameIncluded");
```

Comment:
Room for improvement. Replace "application name" with "module name" if possible. 

Include the module name itself in the output

Reply:
Good spot. If we're including `includeModule` or `includeApplication` here, we can probably just say "name" rather than "application name".